### ISSUE TITLE
Gitea support

### ISSUE THREAD

**Issue by FederAndInk** [gitbutlerapp/gitbutler#2904]
Feature request for [gitea](https://about.gitea.com/) Pull Requests support.

I understand it's still very early, but at least it can be tracked here :)

**Comment by himchawla** [comment]
Hey, are you able to use gitea with gitbutler. Using my private hosted gitea whenever i tried to push it gave me something went wrong, I'll attach a screenshot and maybe logs (if required/ if i can find) tomorrow.

> Feature request for [gitea](https://about.gitea.com/) Pull Requests support.
> 
> I understand it's still very early, but at least it can be tracked here :)

**Comment by techknowlogick** [comment]
This would be great! I’m a part of the Gitea project and would love to help from our side of things (code, debugging, or otherwise). I’m also trialing a new bounty platform, and I think this would be a good candidate. Gitbutler maintainers: would you be ok if I posted a bounty from algora.io?

**Comment by ryanm101** [comment]
Did you check your repo to see if the push did happen? I get that sometimes on Github, but when i look the commit has actually been pushed

**Comment by questmaster** [comment]
Yes, I checked. It didn't work. Nothing pushed.

**Comment by techknowlogick** [comment]
> This would be great! I’m a part of the Gitea project and would love to help from our side of things (code, debugging, or otherwise). I’m also trialling a new bounty platform, and I think this would be a good candidate. Gitbutler maintainers: would you be ok if I posted a bounty from algora.io?

With @schacon using [algora.io for zed](https://twitter.com/chacon/status/1770005036170375594), I hope posting one here is fine (if not, please delete the post from the algora bot and I can delete it from my side too).

**Comment by Rutik7066** [comment]
@FederAndInk I would like to work on this issue. Could you please assign this.

**Comment by himchawla** [comment]
Using this new option works like a charm,

![image](https://github.com/gitbutlerapp/gitbutler/assets/58020143/e0a310ff-07ff-427c-89a0-f37044ea4f1e)

And having test credentials option right there is such a convinience, thanks a bunch gitbutler team.

![image](https://github.com/gitbutlerapp/gitbutler/assets/58020143/17b3e069-5cd6-44de-b6a1-7f04f65e6878)

**Comment by syedbarimanjan** [comment]
@Byron can i work on this.

**Comment by Byron** [comment]
Thanks for offering! GitLab support was supposed to be added first, which would certainly bring some abstractions to make integrations a little more pluggable. Thus, I think it's best to wait on #2511 .

**Comment by mtsgrd** [comment]
Hey guys, I recently landed some changes in #4315 that brings us a bit closer to a solution. There is a now a [set of interfaces](https://github.com/gitbutlerapp/gitbutler/tree/master/app/src/lib/gitHost/interface) that could be implemented to support Gitea, with a factory that can provide the right implementation on a per repo basis.
https://github.com/gitbutlerapp/gitbutler/blob/5fb79bd15da8bc8806f17f9c81443da21813236d/app/src/lib/gitHost/gitHostFactory.ts#L13-L23
That said, we are still lacking support for multiple oauth flows in the backend. Stay tuned!

**Comment by fraschm1998** [comment]
Has this been implemented yet?

**Comment by aman5874** [comment]
@Byron can I work on this?

**Comment by schacon** [comment]
@aman5874 GitLab integration was recently added, which would be a good template for Gitea support.

**Comment by drainpixie** [comment]
I would like to work on this, /attempt #2904

**Comment by OscarLundberg** [comment]
I did some work on this #9170 but it's not complete. probably wont do more work on it, but mentioning it in case someone wants to pick it up.

**Comment by AkihitoNarihisago** [comment]
> I did some work on this [#9170](https://github.com/gitbutlerapp/gitbutler/pull/9170) but it's not complete. probably wont do more work on it, but mentioning it in case someone wants to pick it up.

so this is still open ?

**Comment by Tritonn204** [comment]
/attempt https://github.com/gitbutlerapp/gitbutler/issues/2904

**Comment by Tritonn204** [comment]
I'll have a crack at this over the next while. Is it pretty much push/pull/merge/PR support, or creating repos as well?

**Comment by saimuren** [comment]
I'd like to implement Gitea support. I'm familiar with both the Gitea API and GitButler's integration architecture.

### Implementation Plan
1. **API Client**: Create `gitea.rs` with Gitea REST API bindings (repos, issues, PRs)
2. **Authentication**: OAuth2 + Personal Access Token support
3. **Integration Points**:
   - Repository browsing and cloning
   - PR creation and management
   - Issue linking
   - Webhook support for real-time updates
4. **UI**: Add Gitea provider option in settings, following existing GitHub/GitLab patterns

The Gitea API is largely compatible with GitHub's, so much of the existing GitHub integration code can be adapted. Ready to start.

**Comment by Byron** [comment]
Thanks for offering.
I think it would make sense if you were to mostly hand-roll the implementation. Something I'd want to avoid is to receive thousands of lines of code which aren't reviewable. Any PR submitted to add Gitea support should be small and focussed, maybe starting with just authentication, with PR linking as the first actual feature.
Thanks for your understanding.

**Comment by Excellencedev** [comment]
@Byron Still can't run this app

**Comment by yuliuyi717-ux** [comment]
/attempt #2904

I can take this bounty.

Implementation plan (incremental, review-friendly):
1) Start with Gitea authentication plumbing only (OAuth/token flow), matching the maintainers' request for small, focused PRs.
2) Implement a minimal Gitea git host provider behind existing host abstractions (no broad refactors), then wire repository detection + basic host metadata.
3) Add PR linking/creation flow as the first functional milestone, using existing GitHub/GitLab patterns where possible.
4) Follow with narrow, additive steps for issue/PR status sync and edge-case handling (self-hosted URLs, auth errors, rate limits).
5) Add tests for auth + host resolution + PR linking behavior in each PR slice.

I will keep each PR small and independently reviewable.

**Comment by tribhuwan-kumar** [comment]
@Byron 

i'd like to work on this, could you assign this to me?  
my implementation would be in separate pr so you could review easily!!

**Comment by Byron** [comment]
Please feel free to submit a PR, even without an assignment.

**Comment by Excellencedev** [comment]
@Byron i want to reattempt this. i just want to know how exactly I should split up the PRs

**Comment by Chetansahney** [comment]
Hi maintainers, can I still attempt it?

**Comment by Byron** [comment]
Sorry for the hassle everyone, but I think the constant noise from this issue has to stop. And to achieve that, I really have to close it.

When the time comes, we may reopen it, but what's much more likely is that we will implement this ourselves. Because nowadays, taking a shot at this is so much easier than actually reviewing what comes out such one-shots.
